### PR TITLE
Filter contacts

### PR DESCRIPTION
**Pull request by cnnrdltn**

### CODE REVIEW

File: Pods/EPContactsPicker.swift [hunk 1]
```diff
@@ -255,160 +260,161 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
...
+      }
+
+      let store = CNContactStore()
+      do {
+        filteredContacts = try store.unifiedContacts(matching: predicate,
+                                                     keysToFetch: allowedContactKeys())
+        //print("\(filteredContacts.count) count")
+        if let contacts = self.contactDelegate?.epContactPicker(self, filter: filteredContacts) {
+          filteredContacts = contacts
         }
```
Comment (cnnrdltn) [inline Pods/EPContactsPicker.swift]:
```suggestion
            filteredContacts = try store.unifiedContacts(matching: predicate,
                                                     keysToFetch: allowedContactKeys())
            let delegateFilteredContacts = filteredContacts.filter { [weak self] contact in
                self?.contactDelegate?.epContactPicker(self, shouldShow: contact)
            }
        }
```

@aly-simplehabit  I don't think we need a second delegate.  We could add a `context` or `mode` parameter to the shouldShow method which we could use to distinguish the `search` and `list` cases if needed.

Either way, we should have a consistent method where we either filter one contact at a time or filter the whole array.

**Review by cnnrdltn — COMMENTED** [review]
minor note

**Comment by cnnrdltn** [comment]
@aly-simplehabit  could you describe the issue you're seeing here?  I'm not sure I understand.  are you not getting the appropriate list when searching?

File: Pods/EPContactsPicker.swift (hunk 1 above)
Comment (aly-simplehabit) [reply Pods/EPContactsPicker.swift]: @cnnrdltn Awesome idea, i don't know why i created new delegate, the only one is enough.

**Comment by aly-simplehabit** [comment]
> @aly-simplehabit could you describe the issue you're seeing here? I'm not sure I understand. are you not getting the appropriate list when searching?

My question now why when returning array from

**Pull request closed without merging**